parallel_cluster/bin/start_cluster.sh

```diff
@@ -117,40 +155,35 @@ fi
 
-# Ensure cluster-name is set
-if [[ "${cluster_name_arg}" ]]; then
-    # Log what's going to be run
-    echo_stderr "Running the following command:"
-    echo_stderr "pcluster create ${pcluster_args}"
-    # Initialise the cluster
-    if ! eval pcluster create "${pcluster_args}"; then
-      echo_stderr "Could not create the parallel cluster stack, exiting"
-      exit 1
-    else
-      echo_stderr "Stack creation succeeded - now retrieving name of IP"
-    fi
-
-    # Output the master IP to log
-    # We list both those in pending/initializing or running states.
-    # And we ensure that this is our instance by specifying the Creator tag
-    instance_id="$(aws ec2 describe-instances \
-                     --query "Reservations[*].Instances[*].[InstanceId]" \
-                     --filters "Name=instance-state-name,Values=pending,running" \
-                               "Name=tag:Name,Values=Master" \
-                               "Name=tag:Creator,Values=\"${USER}\"" \
-                     --output text)"
-
-    if [[ -z "${instance_id}" ]]; then
-      echo_stderr "Could not retrieve instance ID"
-    else
-      echo_stderr "Got instance_id ${instance_id}"
-      echo_stderr "Log into your cluster with \"ssm ${instance_id}\""
-    fi
-
-	  # FIXME: control error codes better, avoiding counterintuitive ones: i.e authed within a different account:
-	  # ERROR: The configuration parameter 'vpc_id' generated the following errors:
-	  # The vpc ID 'vpc-7d2b2e1a' does not exist
-	  # OR ERROR:  The following resource(s) failed to create: [MasterServerWaitCondition, ComputeFleet].
+# Log what's going to be run
+echo_stderr "Running the following command:"
+eval echo "pcluster create ${pcluster_args}" 1>&2
+# Initialise the cluster
+if ! eval pcluster create "${pcluster_args}"; then
+  echo_stderr "Could not create the parallel cluster stack, exiting"
+  exit 1
 else
-    display_help
+  echo_stderr "Stack creation succeeded - now retrieving name of IP"
 fi
+
+# Output the master IP to log
+# We list both those in pending/initializing or running states.
+# And we ensure that this is our instance by specifying the Creator tag
+instance_id="$(aws ec2 describe-instances \
+                 --query "Reservations[*].Instances[*].[InstanceId]" \
+                 --filters "Name=instance-state-name,Values=pending,running" \
+                           "Name=tag:Name,Values=Master" \
+                           "Name=tag:Creator,Values=\"${USER}\"" \
+                 --output text)"
+
+if [[ -z "${instance_id}" ]]; then
+  echo_stderr "Could not retrieve instance ID"
+else
+  echo_stderr "Got instance_id ${instance_id}"
+  echo_stderr "Log into your cluster with \"ssm ${instance_id}\""
+fi
+
+# FIXME: control error codes better, avoiding counterintuitive ones: i.e authed within a different account:
+# ERROR: The configuration parameter 'vpc_id' generated the following errors:
+# The vpc ID 'vpc-7d2b2e1a' does not exist
```

Comment:
Where does that copy-pasted VPC error come from? :-S

Also, the code above makes it seem like that error management is solved?

Reply:
If a dev user runs through the Tothill account, I don't have access to that VPC.

Reply:
I think this is still on the TODO list, or maybe I write something up in the troubleshooting section of the Readme.
I just started the download of the gridss hg38 refdata into an efs system and regretting not prioritising that as a morning task so now looking for things to do...